pkgs/by-name/la/lammps/package.nix

```diff
@@ -44,9 +45,21 @@
   extraCmakeFlags ? { },
   # Extra `buildInputs` - meant for packages that require more inputs
   extraBuildInputs ? [ ],
+  overrideStdenv ? null,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
+  # Ease adding CUDA support. See:
+  # https://github.com/lammps/lammps/issues/4420
+  stdenv' =
+    if !(builtins.isNull overrideStdenv) then
+      overrideStdenv
+    else if (extraCmakeFlags.GPU_API or "") == "cuda" then
```

Comment:
You could make it the CUDA build depended on `config.cudaSupport` , add conditionals in the derivation to configure it accordingly. I.e. this would switch on CUDA support automatically if it is turned on globally.

Reply:
> You could make it the CUDA build depended on `config.cudaSupport` , add conditionals in the derivation to configure it accordingly. I.e. this would switch on CUDA support automatically if it is turned on globally.

Done.